### ISSUE TITLE
make linelist object dplyr compatible

### ISSUE THREAD

**Issue by thibautjombart** [reconhub/linelist#13]
The idea would be to keep the metadata infrastructure consistent when doing a bunch of `dplyr` operations. 
This would involve:

- [x] that a `linelist` can also inherit `tibble`
- [x] make sure metadata are preserver over operations affecting columns (e.g. `select`, `group_by`)
- [x] same, for operations on rows (e.g. `filter`)
- [x] enable the use of `epivars` as arguments of these functions

I guess we'd need to see how attributes are handled for now, and maybe write S3 methods for linelist acting as wrappers around `dplyr`

**Comment by zkamvar** [comment]
In a way, this has been addressed in #37, but it's still a bit wonky since you have to do:

```r
my_linelist %>%
  mask() %>% # basically an auto-rename of the epivars
  group_by(case_definition) %>%
  # some stuff with dplyr
  unmask()
```

**Comment by zkamvar** [comment]
There's also things like `lookup(x, case_definition)`, which will return the column name, but it's a bit complicated to use with dplyr since you have to get into the `!!` stuff.
